src/tools/tests/goldens/generated-schemas.jvm.kt

```diff
@@ -271,6 +273,63 @@ abstract class AbstractGold : BaseParticle() {
         }
     }
 
+    @Suppress("UNCHECKED_CAST")
+    class GoldInternal2 private constructor(
+        amt: Double = 0.0,
+        entityId: String? = null,
+        creationTimestamp: Long = RawEntity.UNINITIALIZED_TIMESTAMP,
+        expirationTimestamp:  Long = RawEntity.UNINITIALIZED_TIMESTAMP
+    ) : EntityBase("GoldInternal2", SCHEMA, entityId, creationTimestamp, expirationTimestamp) {
+
+        var amt: Double
+            get() = super.getSingletonValue("amt") as Double? ?: 0.0
+            private set(_value) = super.setSingletonValue("amt", _value)
+
+        init {
+            this.amt = amt
+        }
+        /**
+         * Use this method to create a new, distinctly identified copy of the entity.
+         * Storing the copy will result in a new copy of the data being stored.
+         */
+        fun copy(amt: Double = this.amt) = GoldInternal2(amt = amt)
```

Comment:
Both copy() and mutate() will need to copy over the values of the fields in the EntityBase (EntityBase.singleton and EntityBase.collections), as we discussed and described in the design doc.

Reply:
I need to double check, but... 
Is a separate code-gen path really required to achieve this requirement? Are we doing this already? 

According the [Kotlin docs](https://kotlinlang.org/docs/reference/properties.html#getters-and-setters), `If we define a custom getter, it will be called every time we access the property (this allows us to implement a computed property)`.

Thus, `this.amt` calls `super.getSingletonValue("amt")`, right? What else would be needed beyond this?

Reply:
Oh... we'd need to copy everything else!

Reply:
Cool, will get to implementing this soon.